businfo/availableBusInformation/service.go

```diff
@@ -0,0 +1,43 @@
+package availableBusInformation
+
+import (
+	"encoding/json"
+	"github.com/fahimimam/busbdChckr/businfo/client"
+	"github.com/fahimimam/busbdChckr/businfo/models"
+	"net/http"
+)
+
+type RequestPld struct {
+	Date          string `json:"date"`
+	Identifier    string `json:"identifier"`
+	StructureType string `json:"structureType"`
+}
+
+func GetAvailableBusInformation(data RequestPld) (*models.BusInfo, error) {
+	buf, err := BodyBuffer(data)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", "https://api.bdtickets.com:20102/v1/coaches/search", buf)
```

Comment:
So, since this is hard coded inside the function, what's your plan in the event of needing to add more providers or the provider changes URL of the API?